packages/edit-site/src/components/page-library/use-library-settings.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { store as editSiteStore } from '../../store';
+
+export default function useLibrarySettings() {
+	const storedSettings = useSelect( ( select ) => {
+		const { getSettings } = unlock( select( editSiteStore ) );
+		return getSettings();
+	}, [] );
+
+	const settingsBlockPatterns =
+		storedSettings.__experimentalAdditionalBlockPatterns ?? // WP 6.0
+		storedSettings.__experimentalBlockPatterns; // WP 5.9
+
+	const restBlockPatterns = useSelect(
+		( select ) => select( coreStore ).getBlockPatterns(),
+		[]
+	);
+
+	const blockPatterns = useMemo(
+		() =>
+			[
+				...( settingsBlockPatterns || [] ),
+				...( restBlockPatterns || [] ),
+			].filter(
+				( x, index, arr ) =>
+					index === arr.findIndex( ( y ) => x.name === y.name )
+			),
+		[ settingsBlockPatterns, restBlockPatterns ]
+	);
```

Comment:
Might be good to add a brief comment describing what the code does and the variables `x` and `y` could be more descriptive.

It could be that the code was copy/pasted though (I didn't check all the deleted code), in which case it's an existing problem that might be best left to a follow-up code quality PR.

Reply:
This was copy-and-pasted from the setup of the site editor's block editor. I don't have any issues with improving it.

Reply:
Lets leave it till later 👍